src/ray/core_worker/core_worker.cc

```diff
@@ -4015,7 +4019,15 @@ void CoreWorker::HandleExit(rpc::ExitRequest request,
   // any object pinning RPCs in flight.
   bool is_idle = !own_objects && pins_in_flight == 0;
   bool force_exit = request.force_exit();
-  RAY_LOG(DEBUG) << "Exiting: is_idle: " << is_idle << " force_exit: " << force_exit;
+  auto worker_id = WorkerID::FromBinary(request.worker_id());
```

Comment:
Based on your current logic, worker_id is always the current worker I think so you are disconnecting from yourself.

Reply:
I don’t quite understand why. This worker_id is a parameter passed from remote worker. The driver worker will disconnect with it.